phases/03-deep-learning-core/05-loss-functions/code/main.py

```
@@ -0,0 +1,262 @@
+def binary_cross_entropy(predictions, targets, eps=1e-15):
+    n = len(predictions)
+    total = 0.0
+    for p, t in zip(predictions, targets):
+        p_clipped = max(eps, min(1 - eps, p))
+        total += -(t * math.log(p_clipped) + (1 - t) * math.log(1 - p_clipped))
+    return total / n
+
+
+def bce_gradient(predictions, targets, eps=1e-15):
+    grads = []
+    for p, t in zip(predictions, targets):
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)